Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivityViewModel.kt

```diff
@@ -145,16 +150,18 @@ class MainActivityViewModel @AssistedInject constructor(
 
     private suspend fun QrCodeExtractor<CoronaTestQRCode>.handleCoronaTestQr(uriString: String) = try {
         val qrCode = extract(rawString = uriString)
-        val test = submissionRepository.testForType(qrCode.type).first()
-        when {
-            test != null -> event.postValue(MainActivityEvent.GoToDeletionScreen(qrCode))
-            else -> event.postValue(MainActivityEvent.GoToSubmissionConsentFragment(qrCode))
-        }
+        val coronaTestResult = coronaTestQRCodeHandler.handleQrCode(qrCode)
+        mutableCoronaTestResult.postValue(coronaTestResult)
     } catch (e: Exception) {
         Timber.w(e, "onNavigationUri failed")
         event.postValue(MainActivityEvent.Error(e))
     }
 
+    fun restoreCoronaTest(recycledCoronaTest: CoronaTest) = launch {
```

Comment:
As I mentioned above the restoration should happen probably after test selection screen anyway. So it is no needed here